dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java

```diff
@@ -47,39 +61,71 @@ protected PageResource(HTMLPageAssetAPI pageApi, HostAPI hostApi, WebResource we
+        // Force authentication
+        final InitDataObject auth = webResource.init(false, request, false);
+        final User user = auth.getUser();
+        Response res = null;
+        try {
+            final PageView pageView = this.pageResourceHelper.getPageData(request, response,
+                    user, uri);
+            final String json = this.pageResourceHelper.asJson(pageView);
+            final Response.ResponseBuilder responseBuilder = Response.ok(json);
+            responseBuilder.header("Access-Control-Expose-Headers", "Authorization");
+            responseBuilder.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, " +
+                    "Content-Type, " + "Accept, Authorization");
+            res = responseBuilder.build();
+        } catch (JsonProcessingException e) {
+            res = ExceptionMapperUtil.createResponse(null, "An error occurred when generating " +
+                    "the JSON response (" + e.getMessage() + ")");
+        } catch (DotSecurityException e) {
+            res = ExceptionMapperUtil.createResponse(null, "The user does not have the required"
+                    + " permissions (" + e.getMessage() + ")");
+        } catch (DotDataException e) {
+            res = ExceptionMapperUtil.createResponse(null, "An error occurred when accessing " +
+                    "the page information permissions (" + e.getMessage() + ")");
+        } catch (Exception e) {
+            res = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+        return res;
     }
 
-
-    private User getUser(@Context HttpServletRequest request) {
-        return webResource.init(true, request, true).getUser();
+    @NoCache
+    @GET
+    @Path("/render/{uri: .*}")
+    public Response renderPage(@Context HttpServletRequest request, @Context HttpServletResponse
+            response, @PathParam("uri") String uri) throws Exception {
+        // Force authentication
+        final InitDataObject auth = webResource.init(false, request, false);
+        final User user = auth.getUser();
+        Response res = null;
+        try {
+            final PageView pageView = this.pageResourceHelper.getPageDataRendered(request,
+                    response, user, uri);
+            final String json = this.pageResourceHelper.asJson(pageView);
+            final Response.ResponseBuilder responseBuilder = Response.ok(json);
+            responseBuilder.header("Access-Control-Expose-Headers", "Authorization");
+            responseBuilder.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, " +
+                    "Content-Type, " + "Accept, Authorization");
+            res = responseBuilder.build();
+        } catch (JsonProcessingException e) {
+            res = ExceptionMapperUtil.createResponse(null, "An error occurred when generating " +
+                    "the JSON response (" + e.getMessage() + ")");
+        } catch (DotSecurityException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)